massa-bootstrap/src/client.rs

```diff
@@ -267,39 +267,30 @@ async fn bootstrap_from_server(
         Ok(Ok(msg)) => return Err(BootstrapError::UnexpectedServerMessage(msg)),
     };
 
-    let recv_time_uncompensated = MassaTime::now(0)?;
+    // get the time of reception
+    let recv_time = MassaTime::now()?;
 
     // compute ping
-    let ping = recv_time_uncompensated.saturating_sub(send_time_uncompensated);
+    let ping = recv_time.saturating_sub(send_time_uncompensated);
     if ping > cfg.max_ping {
         return Err(BootstrapError::GeneralError(
             "bootstrap ping too high".into(),
         ));
     }
 
-    // compute compensation
-    let compensation_millis = if cfg.enable_clock_synchronization {
-        let local_time_uncompensated =
-            recv_time_uncompensated.checked_sub(ping.checked_div_u64(2)?)?;
-        let compensation_millis = if server_time >= local_time_uncompensated {
-            server_time
-                .saturating_sub(local_time_uncompensated)
-                .to_millis()
-        } else {
-            local_time_uncompensated
-                .saturating_sub(server_time)
-                .to_millis()
-        };
-        let compensation_millis: i64 = compensation_millis.try_into().map_err(|_| {
-            BootstrapError::GeneralError("Failed to convert compensation time into i64".into())
-        })?;
-        debug!("Server clock compensation set to: {}", compensation_millis);
-        compensation_millis
-    } else {
-        0
-    };
+    // compute client / server clock delta
+    let adjusted_server_time = server_time.checked_add(ping.checked_div_u64(2)?)?;
```

Comment:
Why the divide by 2 here?

Reply:
Approximation of the time it took the message to do `server -> client`, the complete ping value being `client -> server -> client`

Reply:
Maybe you can add this explanation in the comment ?